fmriprep/workflows/bold/base.py

```diff
@@ -511,8 +512,10 @@ def init_func_preproc_wf(bold_file):
                 ('outputnode.bold', 'inputnode.in_file')]),
             (bold_t2s_wf, bold_confounds_wf, [
                 ('outputnode.bold', 'inputnode.bold')]),
-            (bold_t2s_wf, bold_t1_trans_wf, [
-                ('outputnode.bold', 'inputnode.bold_split')]),
+            (bold_t2s_wf, split_opt_comb, [
+                ('outputnode.bold', 'in_file')]),
+            (split_opt_comb, bold_t1_trans_wf, [
+                ('out_files', 'inputnode.bold_split')]),
```

Comment:
Then in this function, we want

```Python
bold_t1_trans_wf.inputs.inputnode.fieldwarp = 'identity'
bold_t1_trans_wf.inputs.inputnode.hmc_xforms = 'identity'
```

This will involve only connecting these inputs to workflow inputs in the `if not multiecho` branches, so removing this:

https://github.com/poldracklab/fmriprep/blob/e8cab504fd6a32785e5905c4b76e946ab88e81b9/fmriprep/workflows/bold/base.py#L441-L442

And removing `fieldwarp` from here:

https://github.com/poldracklab/fmriprep/blob/e8cab504fd6a32785e5905c4b76e946ab88e81b9/fmriprep/workflows/bold/base.py#L457-L460

Now I'm curious why we aren't applying fieldmaps for multi-echo, though. I don't see anywhere they're passed into the `bold_t2s_wf`.

Reply:
Doesn't `bold_bold_trans_wf` include both HMC and SDC?

https://github.com/poldracklab/fmriprep/blob/e8cab504fd6a32785e5905c4b76e946ab88e81b9/fmriprep/workflows/bold/base.py#L461-L463

Then `bold_bold_trans_wf` feeds into `skullstrip_bold_wf`.

https://github.com/poldracklab/fmriprep/blob/e8cab504fd6a32785e5905c4b76e946ab88e81b9/fmriprep/workflows/bold/base.py#L511-L512

And `skullstrip_bold_wf` --> `join_echos` --> `bold_t2s_wf`

https://github.com/poldracklab/fmriprep/blob/e8cab504fd6a32785e5905c4b76e946ab88e81b9/fmriprep/workflows/bold/base.py#L404-L409

Reply:
You're right... We should clean up those statements so that this path is a lot easier to find.